### PR TITLE
Support virtual path deployment

### PR DESCRIPTION
**Pull request by liangshiw**
Resolve https://github.com/abpframework/abp/issues/4089

### CODE REVIEW

**Comment by maliming** [comment]
Wait: https://github.com/abpframework/abp/pull/4087

**Comment by maliming** [comment]
For angular developers can refer to this article: http://codebuckets.com/2018/05/26/hosting-angular-in-iis-under-an-existing-website/

**Comment by hikalkan** [comment]
By angular developers, do you mean @mehmet-erim & @armanozak 😄

**Comment by maliming** [comment]
I mean for those developers who deploy angular in IIS sub-applications.  They can refer to the above article to solve the problem.

File: modules/account/src/Volo.Abp.Account.Web/Pages/Account/Register.cshtml.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public virtual async Task<IActionResult> OnPostAsync()
 
             await SignInManager.SignInAsync(user, isPersistent: false);
 
-            return Redirect(ReturnUrl ?? "/"); //TODO: How to ensure safety? IdentityServer requires it however it should be checked somehow!
+            return Redirect(ReturnUrl ?? "~/"); //TODO: How to ensure safety? IdentityServer requires it however it should be checked somehow!
```
Comment (maliming) [inline modules/account/src/Volo.Abp.Account.Web/Pages/Account/Register.cshtml.cs:62]: You can refer to Microsoft's code.

https://github.com/dotnet/aspnetcore/blob/master/src/Identity/UI/src/Areas/Identity/Pages/V4/Account/Login.cshtml.cs#L112

`ReturnUrl = ReturnUrl ?? Url.Content("~/");`

**Comment by liangshiw** [comment]
I added `AbpApplicationPathViewComponent` used to set `abp.appPath` value, but it is not added to the startup template by default. 
For developers who deploy using virtual paths, please add the following code to your module:

```csharp
Configure<AbpLayoutHookOptions>(options =>
{
        options.Add(LayoutHooks.Head.Last,
        typeof(AbpApplicationPathViewComponent));
});
```

**Comment by hanntd** [comment]
Was this issue fixed for the latest abp framework version (7.0.1)? I'm using this version and having this issue when deploying to IIS as virtual application.

**Pull request merged**